x-pack/plugins/apm/common/critical_path/get_critical_path.test.ts

```diff
@@ -0,0 +1,270 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { apm, ApmFields, dedot } from '@kbn/apm-synthtrace';
+import { getWaterfall } from '../../public/components/app/transaction_details/waterfall_with_summary/waterfall_container/waterfall/waterfall_helpers/waterfall_helpers';
+import { Span } from '../../typings/es_schemas/ui/span';
+import { Transaction } from '../../typings/es_schemas/ui/transaction';
+import { getCriticalPath } from './get_critical_path';
+
+describe('getCriticalPath', () => {
+  function getCriticalPathFromEvents(events: ApmFields[]) {
+    const waterfall = getWaterfall(
+      {
+        traceDocs: events.map(
+          (event) => dedot(event, {}) as Transaction | Span
+        ),
+        errorDocs: [],
+        exceedsMax: false,
+        linkedChildrenOfSpanCountBySpanId: {},
+      },
+      events[0]['transaction.id']!
+    );
+
+    return {
+      waterfall,
+      criticalPath: getCriticalPath(waterfall),
+    };
+  }
+  it('adds the only active span to the critical path', () => {
+    const a = apm.service('a', 'development', 'java').instance('a');
```

Comment:
I think I originally planned on having service a and service b. The name doesn't really matter here but I'll change it to `service` because `a` by itself indeed looks a bit odd.